File: BoT/marzbanBot/src/controller/managerCtrl.py
import requests


class Manager:

    def __init__(self, url_prefix):
        self.url_prefix = url_prefix

    def getUsers(self, chat_id):
        response = requests.get(self.url_prefix + f"/manager/get-users?chat_id={chat_id}")
        return response

    def getAdmins(self, chat_id):
        response = requests.get(self.url_prefix + f"/manager/get-admins?chat_id={chat_id}")
        return response

    def assignAdmin(self, chat_id,user_id,admin_id):
        response = requests.patch(self.url_prefix + f"/manager/assign-admin?chat_id={chat_id}"
                                ,data={
                                    "relevantAdmin":admin_id,
                                    "user_chat_id":user_id
                                }
            )
        return response

    def bannUser(self,chat_id,user_id):
        response = requests.patch(self.url_prefix + f"/manager/bann-user?chat_id={chat_id}"
                                  , data={
                                        "user_chat_id": user_id
                                }
                            )

        return response

    def getTemplates(self,chat_id):
        response = requests.get(f"{self.url_prefix}/manager/get-templates?chat_id={chat_id}")
        return response
    def getTemplate(self,chat_id,template_id):
        response = requests.get(f"{self.url_prefix}/manager/get-template?chat_id={chat_id}&template_id={template_id}")
        return response

    def deleteTemplate(self,chat_id,template_id):
        response = requests.delete(f"{self.url_prefix}/manager/delete-template?chat_id={chat_id}",data={"template_id":template_id})
        return response
    def assignTemplate(self,chat_id,user_id,template_id):
        response = requests.post(f"{self.url_prefix}/manager/assign-template?chat_id={chat_id}", data={
            "user_chat_id":user_id,
            "template_id":template_id
        })

        return response
    def deleteAssignedTemplate(self,chat_id,user_id,template_id):
        response = requests.delete(f"{self.url_prefix}/manager/delete-assignment?chat_id={chat_id}", data={
            "user_chat_id":user_id,
            "template_id":template_id
        })

        return response

    def alterUser(self,chat_id,user_id,rule):
        response = requests.patch(f"{self.url_prefix}/manager/alter-user?chat_id={chat_id}",
                                 data={
                                    "user_chat_id":user_id,
                                     "rule":rule
                                 })
        return response

    def getNoneAdmitUsers(self,chat_id):
        response = requests.get(f"{self.url_prefix}/manager/get-none-admit-users?chat_id={chat_id}")

        return response

    def admitNewUser(self,chat_id,user_id):
        response = requests.patch(f"{self.url_prefix}/manager/admit-user?chat_id={chat_id}",data={"user_chat_id":user_id})
        return response

    def getSpamUsers(self,chat_id):
        response = requests.get(f"{self.url_prefix}/manager/get-banned-users?chat_id={chat_id}")

        return response

    def unbannUser(self,chat_id,user_id):
        response = requests.patch(f"{self.url_prefix}/manager/unbann-user?chat_id={chat_id}",data={"user_chat_id":user_id})
        return response

    def addTemplate(self,chat_id,days_limit,data_limit,price,user_limit,in_bounds):
        response = requests.post(f"{self.url_prefix}/manager/add-template?chat_id={chat_id}",data={
            "days_limit":days_limit,
            "data_limit":data_limit,
            "price":price,
            "user_limit":user_limit,
            "in_bound":in_bounds
        })
        return response

File: backend/src/controller/manager.ctrl.js
const request = require('../db/request')



exports.getUsers = async(req) => {
    try{
        let query = `SELECT * FROM KianDB.users WHERE admit =1 and spam=0 and (rule='user' or rule='admin');`

        const users = await request(query,[],req)

        return users

    }catch(err){throw err}
}

exports.getNoneAdmitUsers = async(req) => {
    try{
        let query = `SELECT * FROM KianDB.users WHERE admit =0 ;`

        const users = await request(query,[],req)

        return users

    }catch(err){throw err}
}

exports.getBannedUser = async(req) => {
    try{
        let query = `SELECT * FROM KianDB.users WHERE spam =1 ;`

        const users = await request(query,[],req)

        return users

    }catch(err){throw err}
}

exports.getAdmin = async(req) => {
    try{
        let query = `SELECT * FROM KianDB.users WHERE rule = 'admin';`

        const users = await request(query,[],req)

        return users

    }catch(err){throw err}
}

exports.alterUser = async(values,req) => {
    try{
        let query = `UPDATE KianDB.users
        SET rule= ?
        WHERE chat_id= ? ;`

        await request(query,[values.rule,values.user_chat_id],req)

    }catch(err){throw err}
}

exports.admitUser = async(values,req) => {
    try{
        let query = `UPDATE KianDB.users
            SET admit= 1
            WHERE chat_id= ? ;`

        await request(query,[values.user_chat_id],req)

    }catch(err){throw err}
}



exports.bannUser = async(values,req) => {
    try{
        let query = `UPDATE KianDB.users
            SET spam= 1
            WHERE chat_id= ? ;`

        await request(query,[values.user_chat_id],req)

    }catch(err){throw err}
}

exports.unBannUser = async(values,req) => {
    try{
        let query = `UPDATE KianDB.users
            SET spam= 0
            WHERE chat_id= ? ;`

        await request(query,[values.user_chat_id],req)

    }catch(err){throw err}
}

exports.assignAdmin = async(values,req) => {
    try{
        let query = `UPDATE KianDB.users
            SET relevantAdmin= ?
            WHERE chat_id= ? ;`

        await request(query,[values.relevantAdmin,values.user_chat_id],req)

    }catch(err){throw err}
}

exports.addTemplate = async(values,req) => {
    try{
        let query = `INSERT INTO KianDB.templates
        (days_limit, data_limit, price, user_limit, in_bounds, whoCreated)
        VALUES(?, ?, ?, ?, ?, 'manager');`

        await request(query,[values.days_limit?values.days_limit:null,values.data_limit?values.data_limit:null,values.price,values.user_limit?values.user_limit:null,values.in_bound?values.in_bound:null ],req)

    }catch(err){throw err}
}

exports.deleteTemplate = async(values,req) => {
    try{
        let query = `UPDATE KianDB.templates
        SET isActive=0
        WHERE template_id=?;`

        await request(query,[values.template_id],req)

    }catch(err){throw err}
}

exports.getTemplates = async(req) => {
    try{
        let query = `SELECT DISTINCT days_limit, data_limit, template_id, price, user_limit, in_bounds, whoCreated, parent_template_id
        FROM KianDB.templates
        WHERE isActive = 1 ;`
            

        const templates = await request(query,[],req)
        
        return templates

    }catch(err){throw err}
}

exports.getTemplate = async(values,req) => {
    try{
        let query = `SELECT DISTINCT days_limit, data_limit, template_id, price, user_limit, in_bounds, whoCreated, parent_template_id
        FROM KianDB.templates
        WHERE template_id = ? and isActive=1;`
            
        const template = await request(query,[values.template_id],req)

        return template

    }catch(err){throw err}
}

exports.assignTemplate = async(values,req) => {
    try{
        let query = `INSERT INTO KianDB.user_template
            (chat_id, template_id)
            VALUES(?, ?);`
            

        await request(query,[values.user_chat_id,values.template_id],req)

    }catch(err){throw err}
}

exports.deleteAssignment = async(values,req) => {
    try{
        let query = `DELETE FROM KianDB.user_template
            WHERE chat_id=? AND template_id=?;`
            

        await request(query,[values.user_chat_id,values.template_id],req)

    }catch(err){throw err}
}